### ISSUE TITLE
Call to undefined method SubmissionDAO::getIdsBySetting()

### ISSUE THREAD

**Issue by sourcedump** [pkp/crossrefReferenceLinking#12]
Hello!
running the scheduled task for this plugin, an error prevents correct operation:

```
PHP Fatal error:  Uncaught Error: Call to undefined method SubmissionDAO::getIdsBySetting() in /plugins/generic/crossrefReferenceLinking/CrossrefReferenceLinkingPlugin.inc.php:466
Stack trace:
#0 /plugins/generic/crossrefReferenceLinking/CrossrefReferenceLinkingInfoSender.inc.php(60): CrossrefReferenceLinkingPlugin->getSubmissionsToCheck(Object(Journal))
#1 /lib/pkp/classes/scheduledTask/ScheduledTask.inc.php(146): CrossrefReferenceLinkingInfoSender->executeActions()
#2 /lib/pkp/classes/cliTool/ScheduledTaskTool.inc.php(110): ScheduledTask->execute()
#3 /lib/pkp/classes/cliTool/ScheduledTaskTool.inc.php(94): ScheduledTaskTool->executeTask('plugins.generic...', Array)
#4 /lib/pkp/classes/cliTool/ScheduledTaskTool.inc.php(66): ScheduledTaskTool->parseTasks('in /plugins/generic/crossrefReferenceLinking/CrossrefReferenceLinkingPlugin.inc.php on line 466
```

Do you know something about that? We are using the plugin version 1.0.2.0 and OJS 3.2.1.2.

Thanks!

**Comment by asmecher** [comment]
@sourcedump, the plugin version 1.0.2 is compatible with OJS 3.3.0-x, not 3.2.1-x.

**Comment by sourcedump** [comment]
Thanks @asmecher , my mistake.

**Comment by bolelligallevi** [comment]
Hi @asmecher, @sourcedump tried with both 1.0 and 1.0.1 but they have different problems:
- 1.0: error 'Unrecognized DAO PublishedArticleDAO!'
- 1.0.1: looks not compatible, error PHP Fatal error:  Uncaught Error: Call to undefined method SubmissionDAO::getIdsBySetting() in /plugins/generic/crossrefReferenceLinking/CrossrefReferenceLinkingPlugin.inc.php:466
1.0.1 cames from https://github.com/defstat/crossrefReferenceLinking/tree/pkp/pkp-lib%236071-rlp-master
Can you help?
Best regards

**Comment by asmecher** [comment]
@bolelligallevi, what version of OJS are you using?

**Comment by bolelligallevi** [comment]
@asmecher, is 3.2.1.2 I did not write because was in the post opening the issue (sourcedump and I are working on the same installation).
Thanks and regards

**Comment by asmecher** [comment]
@bolelligallevi, there is no release of the CrossRef reference linking plugin that's compatible with 3.2.x, unfortunately!

**Comment by bolelligallevi** [comment]
@asmecher, the above version history is from the plugin readme…
Is 1.0.1 not working at all?

Version History
---------------
1.0	- Initial Release
1.0.1	- Adapt for OJS 3.2
1.0.2	- Adapt for OJS 3.3

**Comment by asmecher** [comment]
@bolelligallevi, I don't think the version you're calling 1.0.1 is actually what was packaged and pre-released as 1.0.1 by PKP; the 1.0.1 release we packaged is available here:

https://github.com/pkp/crossrefReferenceLinking/releases/tag/v1_0_1-1

...and does not contain any references to the `SubmissionDAO::getIdsBySetting` function that you've noted causing problems. However, it's flagged as a pre-release and so we don't support it (and I'm not sure if it works).

The "definitive" list of plugin versions against OJS/OMP/OPS versions is in the plugin gallery XML:

http://pkp.sfu.ca/ojs/xml/plugins.xml

As you'll see there, there's no release of the crossrefReferenceLinking plugin that supports OJS 3.2.x. YMMV with the pre-release version but you'll need to experiment.
